src/redisjson.rs

```diff
@@ -218,17 +218,25 @@ impl RedisJSON {
         &self,
         path: &str,
         scalar: &str,
-        start: usize,
-        end: usize,
+        start: i64,
+        end: i64,
     ) -> Result<i64, Error> {
         if let Value::Array(arr) = self.get_doc(path)? {
+            if arr.len() <= 0 || end < -1 {
```

Comment:
Why `end < -1`? What about other negative values? It seems like these should be valid.

Reply:
only because of backward compatibility.

Reply:
-1 is a valid value (--> infinity) while smaller than -1 is not a valid value in RedisJSON

Reply:
Ah, I assumed that `-` meant a negative index (as in Python, counting from the end)!
This is a bit weird...